crates/core/src/db/models.rs

```diff
@@ -263,8 +263,22 @@ pub struct File<'a> {
 }
 
 /// A row in the `attributes` table
-#[derive(Debug, Clone, Queryable, Insertable, AsChangeset)]
-pub struct Attribute<'a> {
+#[derive(Debug, Clone, Insertable, AsChangeset)]
+#[table_name = "attributes"]
+pub struct MetadataAttributeWrite<'a> {
+    /// Metadata address
+    pub metadata_address: Cow<'a, str>,
+    /// Attribute name
+    pub name: Option<Cow<'a, str>>,
+    /// Attribute value
+    pub value: Option<Cow<'a, str>>,
+    /// Attribute trait type
+    pub trait_type: Option<Cow<'a, str>>,
+}
```

Comment:
When using generated fields within the db there needs to be a separate write vs read model.

Reply:
The write has insertable and query the Queryable derive.

In my reading on this the docs also suggest not mixing a struct as both queryable and insertable. Thoughts @ray-kast

Reply:
Yeah, for structs where we have write-default fields separate would be good.  Diesel generally recommend making purpose-built models, but I merged them where possible to reduce boilerplate.